### DISCUSSION TITLE
How to porperly add package dependencies?

### DISCUSSION THREAD

**Discussion by borque** [microsoft/winget-pkgs#81897] [Q&A] [answered]
Hello!
I have created an installer for IrfanView PlugIns which is validated by winget an works in sandboxtest.psd1.

As the PlugIns require the main programme IrfanView to be installed in the same version I tried to add a dependency to the manifest. IrfanView is also available here.

I tried adding
`PackageDependencies: IrfanSkiljan.IrfanView`
to my manifest but I always get a warning that this field is unknown.

When I try
`Dependencies: 
 -  PackageDependencies: IrfanSkiljan.IrfanView`

I get the following error:

> Error context: <root>[Dependencies] Description: Value type not permitted by 'type' constraint.
> Error context: <root> Description: Failed to validate against schema associated with property name 'Dependencies'.

The same applies for 
`Dependencies: IrfanSkiljan.IrfanView`

So what am I making wrong?

**Comment by mdanish-kh** [comment, marked as answer]
You can see how they are added in an example manifest [here](https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/PowerAutomateDesktop/2.23.114.22217/Microsoft.PowerAutomateDesktop.installer.yaml#L19)

So, the correct way would be: 
```
Dependencies:
  PackageDependencies:
  - PackageIdentifier: IrfanSkiljan.IrfanView
```
But, as answered by Esco here: https://github.com/microsoft/winget-pkgs/issues/80745#issuecomment-1264345994 , this requires dependency support to be completed which is not the case yet.

**Reply by borque** [reply]
Thanks, this information confirms what I tried after finding another example. It was the third row that I had missed. I was able to complete my manifest and I have submitted it!
